src/FSharp.Core/map.fs

```diff
@@ -748,206 +905,277 @@ type Map<[<EqualityConditionalOn>]'Key, [<EqualityConditionalOn; ComparisonCondi
-            if index + parent.Count > arr.Length then invalidArg "index" "array is smaller than index plus the number of items to copy"
-             
+
+            if index < 0 then
+                invalidArg "index" "index must be positive"
+
+            if index + parent.Count > arr.Length then
+                invalidArg "index" "array is smaller than index plus the number of items to copy"
+
             let mutable i = index
-            for item in parent do 
+
+            for item in parent do
                 arr.[i] <- item.Value
                 i <- i + 1
 
         member _.IsReadOnly = true
 
         member _.Count = parent.Count
-        
+
     interface IEnumerable<'Value> with
         member _.GetEnumerator() =
-            (seq { for item in parent do item.Value}).GetEnumerator()
-            
+            (seq {
+                for item in parent do
+                    item.Value
+            })
+                .GetEnumerator()
+
     interface IEnumerable with
-        member _.GetEnumerator() = 
-            (seq { for item in parent do item.Value }).GetEnumerator() :> IEnumerator
+        member _.GetEnumerator() =
+            (seq {
+                for item in parent do
+                    item.Value
+            })
+                .GetEnumerator()
+            :> IEnumerator
 
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 [<RequireQualifiedAccess>]
-module Map = 
+module Map =
 
     [<CompiledName("IsEmpty")>]
-    let isEmpty (table: Map<_, _>) =
-        table.IsEmpty
+    let isEmpty (table: Map<_, _>) = table.IsEmpty
 
     [<CompiledName("Add")>]
-    let add key value (table: Map<_, _>) =
-        table.Add (key, value)
+    let add key value (table: Map<_, _>) = table.Add(key, value)
 
     [<CompiledName("Change")>]
-    let change key f (table: Map<_, _>) =
-        table.Change (key, f)
+    let change key f (table: Map<_, _>) = table.Change(key, f)
 
     [<CompiledName("Find")>]
-    let find key (table: Map<_, _>) =
-        table.[key]
+    let find key (table: Map<_, _>) = table.[key]
 
     [<CompiledName("TryFind")>]
-    let tryFind key (table: Map<_, _>) =
-        table.TryFind key
+    let tryFind key (table: Map<_, _>) = table.TryFind key
 
     [<CompiledName("Remove")>]
-    let remove key (table: Map<_, _>) =
-        table.Remove key
+    let remove key (table: Map<_, _>) = table.Remove key
 
     [<CompiledName("ContainsKey")>]
-    let containsKey key (table: Map<_, _>) =
-        table.ContainsKey key
+    let containsKey key (table: Map<_, _>) = table.ContainsKey key
 
     [<CompiledName("Iterate")>]
-    let iter action (table: Map<_, _>) =
-        table.Iterate action
+    let iter action (table: Map<_, _>) = table.Iterate action
 
     [<CompiledName("TryPick")>]
-    let tryPick chooser (table: Map<_, _>) =
-        table.TryPick chooser
+    let tryPick chooser (table: Map<_, _>) = table.TryPick chooser
```

Comment:
Interesting this is going back to one-liners.  I'd actually be ok with removing all one-line implementations in FSharp.Core, and always doing multi-line
```fsharp
    let tryPick chooser (table: Map<_, _>) =
        table.TryPick chooser
```

Reply:
I think we would set 
```
fsharp_max_function_binding_width=1
```
to achieve this, whether just in FSharp.Core or all of src

Reply:
I've applied this setting and re-run formatting.